corehq/apps/domain/decorators.py

```diff
@@ -241,6 +241,22 @@ def __inner__(request, domain, *args, **new_kwargs):
     return decorator
 
 
+def api_domain_view(view):
+    """
+    Decorate this with any domain view that should be accessed via api only
+    """
+    @wraps(view)
+    @api_key()
+    @login_and_domain_required
```

Comment:
does `login_and_domain_required` disallow session authentication? if not, this is still vulnerable to CSRF